cobra/flux_analysis/fastcc.py

```diff
@@ -59,28 +100,26 @@ def fastcc(model, flux_threshold=1.0, zero_cutoff=None):
     """
     zero_cutoff = normalize_cutoff(model, zero_cutoff)
 
+    rxns_to_remove = []
+
     with model:
-        obj_vars = []
-        vars_and_cons = []
-        prob = model.problem
-
-        for rxn in model.reactions:
-            var = prob.Variable("auxiliary_{}".format(rxn.id),
-                                lb=0.0, ub=flux_threshold)
-            const = prob.Constraint(rxn.forward_variable +
-                                    rxn.reverse_variable -
-                                    var, name="constraint_{}".format(rxn.id),
-                                    lb=0.0)
-            vars_and_cons.extend([var, const])
-            obj_vars.append(var)
-        model.add_cons_vars(vars_and_cons)
-
-        model.objective = prob.Objective(Zero, sloppy=True, direction="max")
-        model.objective.set_linear_coefficients({v: 1.0 for v in obj_vars})
-
-        sol = model.optimize()
-
-    rxns_to_remove = sol.fluxes[sol.fluxes.abs() < zero_cutoff].index
+        add_fastcc_cons_and_vars(model, flux_threshold)
+
+        # TODO: randomly set to 3; should decide on a default
+        for _ in range(3):
+            sol = model.optimize(objective_sense="max")
+            rxns_to_remove.extend(sol.fluxes[sol.fluxes.abs() <
```

Comment:
This seems odd. Shouln't it be the other way around, meaning you add reactions to a set of non-blocked reactions in each step? So basically something like

```Python
rxns_to_keep.extend(sol.fluxes.index[sol.fluxes.abs() > zero_cutoff])
```